sycl/test-e2e/Matrix/Inputs/joint_matrix_bfloat16_impl.hpp

```diff
@@ -114,35 +134,68 @@ int main() {
 
   for (unsigned int i = 0; i < combinations.size(); i++) {
     if (combinations[i].nsize == 0) { // Intel AMX
-      test<bfloat16, float, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
       break;
     }
 
     if (combinations[i].nsize == 16) { // architecture::intel_gpu_pvc
-      test<bfloat16, float, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
-      // test<bfloat16, bfloat16, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
 
-      // This combination is not currently supported for sub group size = 32 in
-      // IGC
+      // These combinations are not currently supported for sub group size = 32
+      // in IGC
 #if (!defined(SG_SZ) || SG_SZ != 32)
-      test<bfloat16, float, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
-      // test<bfloat16, bfloat16, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
-      test<bfloat16, float, /*TM*/ 1, /*TN*/ 64, /*TK*/ 16>();
-      // test<bfloat16, bfloat16, /*TM*/ 1, /*TN*/ 64, /*TK*/ 16>();
-      test<bfloat16, float, /*TM*/ 32, /*TN*/ 64, /*TK*/ 16>();
-      // test<bfloat16, bfloat16, /*TM*/ 32, /*TN*/ 64, /*TK*/ 16>();
-      // test<bfloat16, float, /*TM*/ 32, /*TN*/ 64, /*TK*/ 32>();
-      // test<bfloat16, bfloat16, /*TM*/ 32, /*TN*/ 64, /*TK*/ 32>();
-      // test<bfloat16, float, /*TM*/ 1, /*TN*/ 64, /*TK*/ 32>();
-      // test<bfloat16, bfloat16, /*TM*/ 1, /*TN*/ 64, /*TK*/ 32>();
+#if (!defined(ACC_BFLOAT16))
+      test<bfloat16, float, float, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 1, /*TN*/ 64, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 32, /*TN*/ 64, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 32, /*TN*/ 64, /*TK*/ 32>();
+      test<bfloat16, float, float, /*TM*/ 1, /*TN*/ 64, /*TK*/ 32>();
+#endif
+#if (defined(ACC_BFLOAT16))
+      // 8x16x16
+      test<bfloat16, bfloat16, float, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, bfloat16, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, bfloat16, bfloat16, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 8, /*TN*/ 16, /*TK*/ 16>();
+
+      // 16x16x16
+      test<bfloat16, bfloat16, float, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, bfloat16, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, bfloat16, bfloat16, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
+      test<bfloat16, float, float, /*TM*/ 16, /*TN*/ 16, /*TK*/ 16>();
```

Comment:
The code is being repeated. it can be simplified with something like:
```
template<typename Tab, typename TAcc, size_t TM, size_t TN, size_t TK>
void test_combo() {
    test<Tab, Tab, TAcc, TM, TN, TK>();
    test<Tab, TAcc, Tab, TM, TN, TK>();
    test<Tab, Tab, Tab, TM, TN, TK>();
    test<Tab, TAcc, TAcc, TM, TN, TK>();
}

}